packages/flutter/lib/src/material/chip.dart

```diff
@@ -1027,8 +1027,8 @@ class _RawChipState extends State<RawChip>
   void initState() {
     assert(widget.onSelected == null || widget.onPressed == null);
     super.initState();
-    setMaterialState(MaterialState.disabled, !widget.isEnabled);
-    setMaterialState(MaterialState.selected, widget.selected);
+    statesController.update(WidgetState.disabled, !canTap);
+    statesController.update(WidgetState.selected, widget.selected);
```

Comment:
I need to use `canTap` instead of `widget.isEnabled` as `InkWell` does its logic on the provided callbacks